game/world/objects/npc.cpp

```diff
@@ -2175,7 +2197,7 @@ void Npc::nextAiAction(AiQueue& queue, uint64_t dt) {
     case AI_StandUpQuick:
       // NOTE: B_ASSESSTALK calls AI_StandUp, to make npc stand, if it's not on a chair or something
       if(interactive()!=nullptr) {
-        if(!setInteraction(nullptr,false)) {
+        if((interactive()->isLadder() && !isPlayer()) || !setInteraction(nullptr,false)) {
```

Comment:
Why player is special case here?

Reply:
If player uses ladder while this event is triggered automatic climbing occurs. 

Probably the check for empty aiqueue in `Interactive::tick` to find out if marvin `aigoto` is used is bad. Maybe check if go2 flag is not `GT_No` is better?

Reply:
> Hm, maybe than `AI_StandUpQuick` should have no effect on ladders?

You mean ignore or push to queue again?  I swapped empty aiqueue with interactive check in `implMove` in playercontrol for marvin `aigoto`. If player is out of dialog range this ends in a deadlock if interaction is not ended.

> In script(G2), `AI_StandUpQuick` is called from dialog related code, and a few spells: `ZS_MagicSleep`, `ZS_Zapped`, `ZS_Pyro` and some else.

Seems to be called every time before a monster is attacked.

> Can you check please what happens in G1, when some npc climbing ladder and sleep spell is used on him?

Npc climbs, walks to target and then falls asleep.

Reply:
> Npc climbs, walks to target and then falls asleep.

In that case enqueue again `AI_StandUpQuick` make most sense:
this will block all other commands until climb animation is finished